index.js

```diff
@@ -31,31 +28,36 @@ class APCAccess {
 
     // The following can't be defined on boot, so define them optionally in config
     this.contactSensor = new Service.ContactSensor(config.name || 'APCAccess UPS');
+    this.contactSensor
+      .getCharacteristic(Characteristic.ContactSensorState)
+      .on('get', this.getContactState.bind(this));
+
     this.informationService = new Service.AccessoryInformation();
     this.informationService
       .setCharacteristic(
         Characteristic.Manufacturer,
         config.manufacturer || 'American Power Conversion',
       )
+      // I see MODEL: 'Smart-UPS 750 ', in the data
       .setCharacteristic(Characteristic.Model, config.model || 'APCAccess UPS')
+
+      // At least for my device I'm seeing a S/N being return in the data
+      // I'm also seeing Firmware version, which can also be set as a characteristic in HK
+      // SERIALNO: 'AS1539123101  ',
+      // FIRMWARE: 'UPS 09.3 / ID=18',
```

Comment:
Yeah, these values can be pulled but would require a rewrite (a la feature sniffing) to defer establishing the accessory until after the first value set is read.

Reply:
Might be a simpler way of solving this. I'll have a think